tidy3d/components/medium.py

```diff
@@ -82,24 +82,122 @@ def _eps_model(self, frequency: float) -> complex:
+                    "'n0' or add sources to the simulation."
+                )
+            n, k = medium.nk_model(freqs[0])
+            return n + 1j * k
+        # if n0 is specified, we should still warn if it isn't consistent with the sources.
+        if len(freqs) == 0:
+            return n0
+        if not np.all(np.isclose(freqs, freqs[0])):
+            log.warning(
+                f"Class '{self}' given 'n0', but the simulation contains sources "
+                "of different central frequencies. For dispersive or conductive materials, "
+                "this could lead to incorrect parameters for the nonlinearity."
+            )
+            return n0
+        n, k = medium.nk_model(freqs[0])
+        nk = n + 1j * k
+        if not np.isclose(nk, n0):
+            log.warning(
+                f"Class '{self}' given 'n0={n0}' but the complex refractive index "
+                f"of the medium at the source central frequency 'freq0={freqs[0]}' is "
+                f"'n + 1j*k = {nk}'. This could lead to incorrect parameters for "
+                "the nonlinearity."
+            )
+        return n0
 
-     Note
+
+class NonlinearSusceptibility(NonlinearModel):
+    """Model for an instantaneous nonlinear chi3 susceptibility.
+    The expression for the instantaneous nonlinear polarization is given below.
+
+    Note
     ----
-    The nonlinear constitutive relation is solved iteratively; it may not converge
-    for strong nonlinearities. Increasing `numiters` can help with convergence.
+    .. math::
+
+        P_{NL} = \\varepsilon_0 \\chi_3 |E|^2 E
+
+    Note
+    ----
+    This model uses real time-domain fields, so :math:`\\chi_3` must be real.
+    For complex fields (e.g. when using Bloch boundary conditions), the nonlinearity
+    is applied separately to the real and imaginary parts, so that the above equation
+    holds when both E and :math:`P_{NL}` are replaced by their real or imaginary parts.
+    The nonlinearity is applied to the real and imaginary components separately since
+    each of those represents a physical field.
+
+    Note
+    ----
+    Different field components do not interact nonlinearly. For example,
+    when calculating :math:`P_{NL, x}`, we approximate :math:`|E|^2 \\approx |E_x|^2`.
+    This approximation is valid when the E field is predominantly polarized along one
+    of the x, y, or z axes.
+
+    Example
+    -------
+    >>> nonlinear_susceptibility = NonlinearSusceptibility(chi3=1)
     """
 
+    chi3: float = pd.Field(
+        0, title="Chi3", description="Chi3 nonlinear susceptibility.", units="um^2 / V^2"
+    )
+
     numiters: pd.PositiveInt = pd.Field(
-        1,
+        None,
         title="Number of iterations",
-        description="Number of iterations for solving nonlinear constitutive relation.",
+        description="Number of iterations for solving nonlinear constitutive relation. "
+        "Deprecated; use 'NonlinearSpec.num_iters' instead.",
```

Comment:
this only applies if this model is added to nonlinearspec.models? It seems if we set `Medium.nonlinearspec = NonlinearSusceptibility` we should use `numiters`, could you explain? And probably a bit more discussion here could help clarify